car.js

```diff
@@ -135,10 +135,28 @@ async function writeStream (stream) {
   return new CarDatastore(reader, writer)
 }
 
+async function traverseBlock (block, get, car, seen = new Set()) {
+  const cid = await block.cid()
+  await car.put(cid, block.encodeUnsafe())
+  seen.add(cid.toString('base64'))
```

Comment:
doesn't work with CIDv0 unfortunately, maybe just a `toString()` here and line 144?

Reply:
so, the way `CID.toString()` works is that it caches the string type it was instantiated with and uses that as the default. this was done for perf and for some backwards compatibility, but it makes it problematic when using `toString()` as a cache key because it isn’t guaranteed to be consistent 😕

Reply:
So what do? https://github.com/ipfs/js-ipfs/pull/2953 doesn't actually work because of this, I'm using your branch but manually edited this fie to remove `'base64'` here and below, otherwise it blows up as soon as it encounters a CIDv0. Do we need to if/else on the version?

Reply:
We can just switch to explicit `‘base58’` instead. I can’t think of anything it would break.